### ISSUE TITLE
KCL: Reversible arc direction

### ISSUE THREAD

**Issue by greg-kcio** [KittyCAD/modeling-app#1534]
Currently KCL can only create arcs counterclockwise, not clockwise.

**Comment by Irev-Dev** [comment]
My 2c is that this shouldn't be a new stb function or even a property, but instead just a negative number ( I know that doesn't work, but I think conforming to math standards for rotation direction is a good idea)

**Comment by greg-kcio** [comment]
That was my intuition at first too:

Counterclockwise example:

```
arc({
       radius: 10,
       angle_start: 45,
       angle_end: 180
     }, %)
```

Clockwise (reversed) example:
```
arc({
       radius: 10,
       angle_start: 45 - 360,
       angle_end: 180 - 360
     }, %)
```

However it isn't compatible with all of the [ArcData signatures](https://github.com/KittyCAD/modeling-app/blob/main/docs/kcl/std.md#arguments-10) we have today since some omit the start and end angles:

```
{
	// The end angle.
	angle_end: number,
	// The start angle.
	angle_start: number,
	// The radius.
	radius: number,
	// The tag.
	tag: string,
} |
{
	// The end angle.
	angle_end: number,
	// The start angle.
	angle_start: number,
	// The radius.
	radius: number,
} |
{
	// The center.
	center: [number, number],
	// The radius.
	radius: number,
	// The tag.
	tag: string,
	// The to point.
	to: [number, number],
} |
{
	// The center.
	center: [number, number],
	// The radius.
	radius: number,
	// The to point.
	to: [number, number],
}
```

Alternatively we could add a property like `direction: "CCW" | "CW"]` to all forms of `ArcData`, and then `angle_start` and `angle_end` would become interchangeable.

**Comment by Irev-Dev** [comment]
Right, hmm

These signatures feel a little speculative. The reason we have so many ways to define a straight line in kcl is because each one is useful to defining a constraint.

i.e. `lineTo([54.5, ZERO], %)` snaps the end of the line to the x axis, `angledLine([myAng, 5], %)` is useful to constrain the angle. And they were added in parallel with the constraints development. I have a feeling these will get cleaned up when we have a good look at constraints for arcs, if not before then.

**Comment by jessfraz** [comment]
yeah i made too many not knowing wtf kurt wanted or needed we will prune

**Comment by greg-kcio** [comment]
I'm happy with keeping the four `ArcData` signatures above, just want to add something like

```
direction: "CCW" | "CW" = "CCW"
``` 

to each one.

**Comment by gserena01** [comment]
Looking in the engine, not all arcs are created equally. There are tangential arcs (defined by radius, center, and angle OR radius, center, and point to go to) and general arcs. Tangential arcs that go to a point choose clockwise or counterclockwise (depending on the point input) in order to maintain first-degree continuity at the start point. Somewhat similarly, tangential arcs defined by an angle and radius can be (at least with the changes in my [current PR](https://github.com/KittyCAD/engine/pull/2034)) clockwise or counterclockwise if the input angle is positive vs. negative. However, changing the sign of the angle WILL change the endpoint of the arc.

general arcs, (with the changes in my [current PR](https://github.com/KittyCAD/engine/pull/2034)) can be drawn clockwise if the input angles are such that the start angle is greater than the end angle. 

With this in mind, how can we add clarity to the users about how to create clockwise vs. counterclockwise arcs?

**Comment by greg-kcio** [comment]
> general arcs, (with the changes in my [current PR](https://github.com/KittyCAD/engine/pull/2034)) can be drawn clockwise if the input angles are such that the start angle is greater than the end angle.

That's what I was looking for, thanks @gserena01!

> With this in mind, how can we add clarity to the users about how to create clockwise vs. counterclockwise arcs?

^ with documentation and examples :)

**Comment by gserena01** [comment]
I'm happy to add additional notes about this in the modeling-api docs. Is there anywhere else we would want to update notes and examples?

**Comment by greg-kcio** [comment]
@gserena01 I'll add a sample of the Zoo logo to https://github.com/KittyCAD/kcl-samples that demonstrates clockwise and counterclockwise arcs.
